python/ray/tune/execution/tune_controller.py

```diff
@@ -106,6 +106,21 @@ def __init__(
 
         # Removed actors
         self._started_actors: Set[TrackedActor] = set()
+
+        # Map of tracked actors -> timestamp
+        # The timestamp is when we requested the stop.
+        # We track these actors here to force a
+        # cleanup after some time (as they might be hanging).
+        # Todo: This timeout logic should be moved into the actor manager.
+        # This map is populated whenever we request an actor stop:
+        #  - Regular STOP decision
+        #  - Removing an actor because its trial REUSEs a different trial's actor
```

Comment:
I never understood why we need to swap actors and stop the original one if a trial already has an actor attached. Is there a use case that I'm missing?

Reply:
This is for actor reuse:

- We want to start 4 trials, but only 1 can run in parallel
- We request resources for 4 actors to potentially trigger autoscaling
- However, autoscaling either takes a long time or is disabled
- Trial 1 finishes and we cache the actor.
- We now want to start trial 2. We have a cached actor, so we use that
- The actor we originally requested for trial 2 is now not needed anymore (as trial 2 uses the cached actor), so we remove it

Generally actor reuse is faster than removing the old actor and starting a new one. This is especially important for short running trials, trials that are paused frequently, or when the Actor setup (e.g. imports) takes a long time.

Let me know if that makes sense!

Reply:
Oh, got it, the original actor has *already* been stopped. "Removing an actor" in this case means removing the actor metadata (in the form of a `TrackedActor`). I thought it meant stopping the original actor. Then, we'll reuse a live cached actor.